### PR TITLE
[FLINK-32655][runtime] Fix checkpoint aborted message being swallowed by RecreateOnResetOperatorCoordinator

### PR DESCRIPTION
**Pull request by liming30**
… by RecreateOnResetOperatorCoordinator.

## What is the purpose of the change

 Fix checkpoint aborted message being swallowed by `RecreateOnResetOperatorCoordinator`

## Brief change log

Let the `RecreateOnResetOperatorCoordinator` override `notifyCheckpointAborted` method and forward the checkpoint aborted message to the real `OperatorCoordinator` for processing.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by 1996fanrui — COMMENTED** [review]
Hi @liming30 , thanks a lot for this fix, the change LGTM.

Could you help add a test at `RecreateOnResetOperatorCoordinatorTest` to test whether the `notifyCheckpointAborted` is expected?

**Comment by liming30** [comment]
> Could you help add a test at `RecreateOnResetOperatorCoordinatorTest` to test whether the `notifyCheckpointAborted` is expected? Or improve an old unit test.

@1996fanrui Thanks for your review. I added `RecreateOnResetOperatorCoordinatorTest#testNotifyCheckpointAbortedSuccess` to verify that `notifyCheckpointAborted` can be forwarded to the real `OperatorCoordinator`.

File: flink-runtime/src/test/java/org/apache/flink/runtime/operators/coordination/RecreateOnResetOperatorCoordinatorTest.java [hunk 1]
```diff
@@ -264,6 +264,19 @@ public void testConsecutiveResetToCheckpoint() throws Exception {
...
+    public void testNotifyCheckpointAbortedSuccess() throws Exception {
+        TestingCoordinatorProvider provider = new TestingCoordinatorProvider(null);
+        MockOperatorCoordinatorContext context =
+                new MockOperatorCoordinatorContext(OPERATOR_ID, NUM_SUBTASKS);
+        RecreateOnResetOperatorCoordinator coordinator = createCoordinator(provider, context);
+        TestingOperatorCoordinator internalCoordinatorAfterReset =
+                getInternalCoordinator(coordinator);
+
+        coordinator.notifyCheckpointAborted(1L);
+        assertThat(internalCoordinatorAfterReset.getLastCheckpointAborted()).isEqualTo(1L);
```
Comment (1996fanrui) [inline flink-runtime/src/test/java/org/apache/flink/runtime/operators/coordination/RecreateOnResetOperatorCoordinatorTest.java:277]: nits:

```suggestion
        assertThat(internalCoordinatorAfterReset.getLastCheckpointAborted()).isOne();
```

Comment (1996fanrui) [inline flink-runtime/src/test/java/org/apache/flink/runtime/operators/coordination/RecreateOnResetOperatorCoordinatorTest.java:277]: @liming30 , thanks for the update,  I left a minor comment.

nits:

```suggestion
        long checkpointId = 10L;
        coordinator.notifyCheckpointAborted(checkpointId);
        assertThat(internalCoordinatorAfterReset.getLastCheckpointAborted()).isEqualTo(checkpointId);
```

Comment (liming30) [reply flink-runtime/src/test/java/org/apache/flink/runtime/operators/coordination/RecreateOnResetOperatorCoordinatorTest.java]: @1996fanrui thanks, I have updated this part of the code.

**Review by 1996fanrui — APPROVED** [review]
Thanks for the fix, LGTM.

Let's wait for the CI green.

**Comment by 1996fanrui** [comment]
@liming30 , would you mind backporting this fix to 1.16 and 1.17?

**Comment by liming30** [comment]
> @liming30 , would you mind backporting this fix to 1.16 and 1.17?

@1996fanrui Sure, I will initiate a related PR for backporting later. Thanks.

**Comment by 1996fanrui** [comment]
@liming30   please run this command to fix checkstyle.

`mvn spotless:apply`

**Comment by liming30** [comment]
@flinkbot run azure

**Comment by 1996fanrui** [comment]
Thanks for the contribution, merging~

**Pull request merged**